test/unordered/constructor_tests.cpp

```diff
@@ -317,66 +329,208 @@ namespace constructor_tests {
     }
 
 #if !defined(BOOST_NO_CXX11_HDR_INITIALIZER_LIST)
-    std::initializer_list<typename T::value_type> list;
+    typedef typename T::value_type value_type;
+
+    std::initializer_list<value_type> list;
+
+    test::random_values<T> v(3, generator);
+    std::vector<value_type> vec(v.begin(), v.end());
+    BOOST_ASSERT(vec.size() >= 3);
+
+    std::initializer_list<value_type> nonempty_list{vec[0], vec[1], vec[2]};
+
+    // create a new vector here because erase() requires assignability which is
+    // deleted for some of the test types
+    //
+    std::vector<value_type> expected(vec.begin(), vec.begin() + 3);
 
     UNORDERED_SUB_TEST("Initializer list construct 1")
     {
       test::check_instances check_;
 
-      T x(list);
-      BOOST_TEST(x.empty());
-      BOOST_TEST_EQ(x.bucket_count(), 0u);
-      BOOST_TEST(test::equivalent(x.hash_function(), hf));
-      BOOST_TEST(test::equivalent(x.key_eq(), eq));
-      BOOST_TEST(test::equivalent(x.get_allocator(), al));
+      {
+        T x(list);
+        BOOST_TEST(x.empty());
+        BOOST_TEST_EQ(x.bucket_count(), 0u);
+        BOOST_TEST(test::equivalent(x.hash_function(), hf));
+        BOOST_TEST(test::equivalent(x.key_eq(), eq));
+        BOOST_TEST(test::equivalent(x.get_allocator(), al));
+      }
+
+      {
+        T x(nonempty_list);
```

Comment:
This should probably be `T x{ vec[0], vec[1], vec[2] };` instead.

Reply:
This is a nice and simple refactor; will do!